frog/hardware/http_device.py

```diff
@@ -0,0 +1,66 @@
+"""Provides a device which communicates via HTTP requests."""
+
+from abc import abstractmethod
+from collections.abc import Callable
+from functools import partial
+from typing import Any
+
+from PySide6.QtNetwork import QNetworkAccessManager, QNetworkReply, QNetworkRequest
+
+from frog.config import DEFAULT_HTTP_TIMEOUT
+from frog.hardware.device import AbstractDevice
+
+
+class HTTPDevice(
+    AbstractDevice,
+    parameters={
+        "timeout": "The maximum time in seconds to wait for a response from the server"
+    },
+):
+    """A device which communicates via HTTP requests."""
+
+    def __init__(self, timeout: float = DEFAULT_HTTP_TIMEOUT) -> None:
+        """Create a new HTTPDevice.
+
+        Args:
+            timeout: The maximum time in seconds to wait for a response from the server
+        """
+        if timeout <= 0:
```

Comment:
It doesn't look like we enforced this before. I guess we don't want to allow for no timeout? Likely to only cause problems I suppose

Reply:
Yeah, I'm not sure if there'd be a legitimate reason for that. Users can always set a really long timeout if they want